### ISSUE TITLE
Pressing 'o' doesn't open Omnibar

### ISSUE THREAD

**Issue by tommyalatalo** [gdh1995/vimium-c#437]
I just noticed that suddenly my omnibar doesn't open when I press `o`. This worked just a week or two ago when I was setting up vimium-c, so I'm curious if I've missed a setting somewhere or if there has been an update that might have broken something?

`Shift+T` to open the tab selector popup works just fine.

How do I go about troubleshooting this?

**Comment by gdh1995** [comment]
what's your custom key mappings and exclusion rules?



---Original---
From: "Tommy ***@***.***&gt;
Date: Thu, Sep 16, 2021 14:36 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [gdh1995/vimium-c] Pressing 'o' doesn't open Omnibar (#437)




 
I just noticed that suddenly my omnibar doesn't open when I press o. This worked just a week or two ago when I was setting up vimium-c, so I'm curious if I've missed a setting somewhere or if there has been an update that might have broken something?
 
Shift+T to open the tab selector popup works just fine.
 
How do I go about troubleshooting this?
 
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tommyalatalo** [comment]
Here is my complete config: http://hastebin.com/oporelucoh.json

**Comment by gdh1995** [comment]
Maybe your keyboard layout affects "O".&nbsp; You may make another key like Shift+T to vomnibar, and test whether it can show or not.


If not, then there may be a bug.&nbsp; Please test some commands like Vomnibar.activateHistory and *Bookmarks.



---Original---
From: "Tommy ***@***.***&gt;
Date: Thu, Sep 16, 2021 14:53 PM
To: ***@***.***&gt;;
Cc: "Dahan ***@***.******@***.***&gt;;
Subject: Re: [gdh1995/vimium-c] Pressing 'o' doesn't open Omnibar (#437)




 
Here is my complete config: http://hastebin.com/oporelucoh.json
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tommyalatalo** [comment]
Is the default command to open vomnibar just "vomnibar"?
I'll investigate this a bit more later today.

**Comment by gdh1995** [comment]
A basic name is Vomnibar.activate. You may open Vimium C Options, press "?" to see all available commands.



---Original---
From: "Tommy ***@***.***&gt;
Date: Thu, Sep 16, 2021 16:16 PM
To: ***@***.***&gt;;
Cc: "Dahan ***@***.******@***.***&gt;;
Subject: Re: [gdh1995/vimium-c] Pressing 'o' doesn't open Omnibar (#437)




 
Is the default command to open vomnibar just "vomnibar"?
 I'll investigate this a bit more later today.
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tommyalatalo** [comment]
Okay, there definitely seems to be an issue here, and it's something specific to Firefox.
I tried binding the vomnibar to something else, shift+P and shift+T, neither of them worked.

Then I tried loading the same configuration that I have stored for Firefox into Brave instead, there the vomnibar works perfectly.
Furthermore I noticed that on Firefox it seems like vimium stops working entirely after pressing the vomnibar binding, scrolling and the tab menu with shift+T stop working as well as the link hints. I need to reload the page to get vimium up and running again.

I'm on Firefox 92.0 and vimium c 1.92.4.

**Comment by gdh1995** [comment]
When you found it broke, you might use Escape to force any mode to exit itself.



---Original---
From: "Tommy ***@***.***&gt;
Date: Thu, Sep 16, 2021 23:36 PM
To: ***@***.***&gt;;
Cc: "Dahan ***@***.******@***.***&gt;;
Subject: Re: [gdh1995/vimium-c] Pressing 'o' doesn't open Omnibar (#437)




 
Okay, there definitely seems to be an issue here, and it's something specific to Firefox.
 I tried binding the vomnibar to something else, shift+P and shift+T, neither of them worked.
 
Then I tried loading the same configuration that I have stored for Firefox into Brave instead, there the vomnibar works perfectly.
 Furthermore I noticed that on Firefox it seems like vimium stops working entirely after pressing the vomnibar binding, scrolling and the tab menu with shift+T stop working as well as the link hints. I need to reload the page to get vimium up and running again.
 
I'm on Firefox 92.0 and vimium c 1.92.4.
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tommyalatalo** [comment]
Yep, that does seem to bring the focus out of the broken mode, which is the vomnibar.

**Comment by gdh1995** [comment]
Sorry there may be a bug of Vimium C, but it's not clear yet what triggered it. Please help me do some tests:

1. On a normal website, press `o` to show Vomnibar, and wait a while to ensure it fails
2. paste `about:debugging#/runtime/this-firefox` into address bar and open it
3. find the card of Vimium C and click `Inspect` at its top right corner
4. in the opened "Toolbox" tab, click `Console` panel
5. it should have reported some errors in red color, please copy and paste all of log text here.

**Comment by tommyalatalo** [comment]
This is the output from when I press 'o', I think specifically the `e is undefined` error appears when activating the vomnibar. 

```
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. index.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 2 background.js:3557
    send moz-extension://6b2954aa-5ad2-4f77-812b-87ef0dc1c52a/background.js:3557
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. popup.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 4 background.js:3557
    send moz-extension://6b2954aa-5ad2-4f77-812b-87ef0dc1c52a/background.js:3557
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. index.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 2 background.js:3557
    send moz-extension://6b2954aa-5ad2-4f77-812b-87ef0dc1c52a/background.js:3557
e is undefined browsing_data_manager.js:1
    ai moz-extension://f87c2605-60f2-4724-ae1a-d81713d752a0/background/browsing_data_manager.js:1
    apply self-hosted:2736
    applySafeWithoutClone resource://gre/modules/ExtensionCommon.jsm:626
    wrapPromise resource://gre/modules/ExtensionCommon.jsm:855
    withLastError resource://gre/modules/ExtensionCommon.jsm:774
    wrapPromise resource://gre/modules/ExtensionCommon.jsm:843
Unchecked lastError value: Error: An unexpected error occurred browsing_data_manager.js:1
    ai moz-extension://f87c2605-60f2-4724-ae1a-d81713d752a0/background/browsing_data_manager.js:1
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. popup.html
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. index.html
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. popup.html
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. index.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 4 background.js:3557
    send moz-extension://6b2954aa-5ad2-4f77-812b-87ef0dc1c52a/background.js:3557
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. index.html
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. popup.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 4 background.js:3557
    send moz-extension://6b2954aa-5ad2-4f77-812b-87ef0dc1c52a/background.js:3557
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. panel.html
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. popup.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 12 background.js:3557
    send moz-extension://6b2954aa-5ad2-4f77-812b-87ef0dc1c52a/background.js:3557
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. popup.html
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 26 background.js:3557
    send moz-extension://6b2954aa-5ad2-4f77-812b-87ef0dc1c52a/background.js:3557
```

**Comment by gdh1995** [comment]
```
e is undefined browsing_data_manager.js:1
    ai moz-extension://f87c2605-60f2-4724-ae1a-d81713d752a0/background/browsing_data_manager.js:1
```

These 2 lines are very useful. Thank you.
